ls1intum/Artemis
# [Exam] Show start cover in grace period after browser reload

### Checklist

### Motivation and Context


See https://github.com/ls1intum/Artemis/issues/2795. Exercises for the student exam are only fetched after the student confirmed his participation. When a student reloads the page while in the grace period, the end cover for the final submission of the exam is shown to the student currently. The exam submission will fail.

### Description


- [x] Show start cover if exam is visible, grace period is not over, the student has not submitted or confirmed the exercise start (resets on reload) yet
- [x] Add check to only show the end cover if the student has confirmed the start before. Additionally, we show the end cover page if the grace period is over and the student did not submit to display the message that he failed to submit.
- [x] Reduce code duplication and remove useless flag.

### Steps for Testing

**In depth code review if you are a developer. Double check that the conduction timeline (visible, start, grace period, end, summary) and the displayed components in _exam-participation.component_ are non-overlapping and make sense**

1. Log in to Artemis
2. Navigate to Course Administration
3. Create an exam with multiple users
4. Participate in the exam and test hand-in early, regular hand-in, reloading the page, reloading the page in the grace period and submitting afterwards.
5. Check that the summary page is correctly displayed after submitting the exam early or regularly
6. Also try not to submit the exam in the end. A message that you have failed to submit should be shown.
